### PR TITLE
Show better error messages

### PR DESCRIPTION
**Pull request by leo**
As discussed with @rauchg, this PR aims to correctly display any error messages we could be receiving from our APIs! 😏 🌷

### CODE REVIEW

File: src/providers/sh/util/error.js [hunk 1]
```diff
@@ -64,7 +64,12 @@ async function responseError(res) {
     userError = false
   }
 
-  const err = new Error(message || 'Response error')
+  if (!message && fallbackMessage) {
```
Comment (rauchg) [inline src/providers/sh/util/error.js]: `message == null` would be a bit safer

File: src/providers/sh/util/error.js [hunk 2]
```diff
@@ -64,7 +64,12 @@ async function responseError(res) {
...
+    message = fallbackMessage
+  }
+
+  const err = new Error(message ? `${message} (${res.status})` : 'Response error')
```
Comment (rauchg) [inline src/providers/sh/util/error.js]: When would we fall back to `Response error` ? seems like never, and it's too generic and confusing

Comment (leo) [reply src/providers/sh/util/error.js]: Fixed!

File: src/providers/sh/util/error.js (hunk 1 above)
Comment (leo) [reply src/providers/sh/util/error.js]: Fixed!

**Comment by leo** [comment]
[Everything fixed!](https://circleci.com/workflow-run/b7448459-3753-423a-92c1-cde543d82130)

**Comment by leo** [comment]
Building tests for this before it gets merged!

**Comment by leo** [comment]
Tests added! ✅

**Comment by sergiodxa** [comment]
everything looks fine ✅

**Pull request merged**